erpnext/stock/doctype/item_price/item_price.py

```diff
@@ -26,8 +26,10 @@ def validate_item(self):
 			frappe.throw(_("Item {0} not found.").format(self.item_code))
 
 	def validate_dates(self):
+		from frappe.utils import getdate
+		
 		if self.valid_from and self.valid_upto:
-			if self.valid_from > self.valid_upto:
+			if self.valid_from > getdate(self.valid_upto):
```

Comment:
```suggestion
			if getdate(self.valid_from) > getdate(self.valid_upto):
```

Might as well fix this now 😄